Gems/AtomTressFX/Code/Rendering/HairCommon.h

```diff
@@ -0,0 +1,208 @@
+            {
+            public:
+                TYPE* operator->()
+                {
+                    return &m_cpuBuffer;
+                }
+
+                TYPE* get()
+                {
+                    return &m_cpuBuffer;
+                }
+
+                Render::SrgBufferDescriptor& GetBufferDescriptor() { return m_bufferDesc;  }
+
+                //! Use this method only if the buffer will be attached to a single Srg.
+                //! If this is not the case use InitForUndefinedSrg
+                bool InitForUniqueSrg(
+                    Data::Instance<RPI::ShaderResourceGroup> srg,
+                    Render::SrgBufferDescriptor& srgDesc)
+                {
+                    m_srg = srg;
+                    m_bufferDesc = srgDesc;
+
+                    AZ_Error("HairUniformBuffer", m_srg, "Critical Error - no Srg was provided to bind buffer to [%s]",
+                        m_bufferDesc.m_bufferName.GetCStr());
+
+                    RHI::ShaderInputConstantIndex indexHandle = m_srg->FindShaderInputConstantIndex(m_bufferDesc.m_paramNameInSrg);
+                    if (indexHandle.IsNull())
+                    {
+                        AZ_Error("HairUniformBuffer", false,
+                            "Failed to find shader constant buffer index for [%s] in the SRG.",
+                            m_bufferDesc.m_paramNameInSrg.GetCStr());
+                        return false;
+                    }
+
+                    m_bufferDesc.m_resourceShaderIndex = indexHandle.GetIndex();
+                    return true;
+                }
+
+                //! Updates Binds the data to the Srg and copies it to the GPU side.
+                //! Assumes that the buffer is uniquely attached to the saved srg.
+                bool UpdateGPUData()
+                {
+                    if (!m_srg.get())
+                    {
+                        AZ_Error("HairUniformBuffer", false, "Critical Error - no Srg was provided to bind buffer to [%s]",
+                            m_bufferDesc.m_bufferName.GetCStr());
+                        return false;
+                    }
+
+                    RHI::ShaderInputConstantIndex constantIndex = RHI::ShaderInputConstantIndex(m_bufferDesc.m_resourceShaderIndex);
+                    if (constantIndex.IsNull())
+                    {
+                        AZ_Error("HairUniformBuffer", false, "Critical Error - Srg index is not valide for [%s]",
+                            m_bufferDesc.m_paramNameInSrg.GetCStr());
+                        return false;
+                    }
+
+                    if (!m_srg->SetConstantRaw(constantIndex, &m_cpuBuffer, m_bufferDesc.m_elementSize))
+                    {
+                        AZ_Error("HairUniformBuffer", false, "Failed to bind Constant Buffer for [%s]", m_bufferDesc.m_bufferName.GetCStr());
+                        return false;
+                    }
+                    return true;
+                }
```

Comment:
Code duplication: Why not have this function call the 
`bool UpdateGPUData(Data::Instance<RPI::ShaderResourceGroup> srg, Render::SrgBufferDescriptor& srgDesc)`
function below?